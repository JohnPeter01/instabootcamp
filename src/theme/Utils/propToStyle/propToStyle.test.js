const { propToStyle } = require('.');

describe('propToStyle()', () => {
  describe('when recives an simple argument', () => {
    test('and it is a string', () => {
      const propToStyleResult = propToStyle('textAlign');
      const component = { textAlign: 'center' };
      const styledResult = propToStyleResult(component);

      expect(styledResult).toEqual({ textAlign: 'center' });
    });
    test('and it is a number', () => {
      const propToStyleResult = propToStyle('flex');
      const component = { flex: 1 };
      const styledResult = propToStyleResult(component);
      expect(styledResult).toEqual({ flex: 1 });
    });
  });
  describe('renders only one breakpoint resolution', () => {
    test('renders only the recived ones', () => {
      const propToStyleResult = propToStyle('textAlign');
      const component = { textAlign: { xs: 'center' } };
      const styledResult = propToStyleResult(component);

      expect(styledResult).toMatchSnapshot();
    });
    test('renders two or more breakpoints resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      const component = { textAlign: { xs: 'center', md: 'right' } };
      const styledResult = propToStyleResult(component);

      expect(styledResult).toMatchSnapshot();
    });
  });
});
