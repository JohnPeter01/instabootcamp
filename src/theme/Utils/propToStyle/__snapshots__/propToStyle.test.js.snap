// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() renders only one breakpoint resolution renders only the recived ones 1`] = `
Array [
  Array [
    "@media only screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`propToStyle() renders only one breakpoint resolution renders two or more breakpoints resolution 1`] = `
Array [
  Array [
    "@media only screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media only screen and (min-width:",
    "768",
    "px){",
    "text-align: right;",
    "}",
  ],
]
`;
